### ISSUE TITLE
Resolution Helper for CodeCod

### ISSUE THREAD

**Issue by gunnargestsson** [Microsoft/AL#4671]
**Is your feature request related to a problem? Please describe.**
During assignment from a variable to another you might get a hint from the CodeCop of a possible overflow.

![image](https://user-images.githubusercontent.com/23558797/53681616-ed6b8480-3ce3-11e9-8f57-30dd48d5704a.png)

**Describe the solution you'd like**
The Resolution Helper can suggest adding a CopyStr and MaxStrLen like this.
`TempItem.ID := CopyStr(ResultXMLNode.AsXmlElement().InnerText(),1,MaxStrLen(TempItem.ID));`

**Comment by KristofKlein** [comment]
Hm, just thinking .... this is one solution to the "problem". And with all respect, a questionable one. If a  developer with absolute no experiences stumbles on this one and sees this solution he will start "cutting" everything: Names, Addresses, primary key values?! (just remember: please out a commit...) I'm not sure what I should think about the CodeCop rule in question: In fact I like it that I get it, on the other hand, if you suggest to cut the value ... you will experience that something goes wrong on runtime way too late as it would silently produce bad data...I am a fan of "fail fast". so I ignore that rule - treat it as a info that here is code that I have to rewrite at one point (or maybe not!). But I would rather see it crashing in runtime than producing nonsense :)

Long story short: solution suggestion is a nice idea... but i guess not in this case :)

**Comment by NKarolak** [comment]
I agree with Gunnar that the current CodeCop text "Possible overflow assigning ... to ..." could be improved. But I also agree with Kristof that we did not find the optimal solution yet. Though I cannot make a good suggestion myself 💁‍♀️ Maybe something like "_If appropriate, use "_ + Gunnar's suggestion.

BTW, is the documentation outdated, or is it Gunnar's AL extension? According to the [article](https://docs.microsoft.com/en-us/dynamics365/business-central/dev-itpro/developer/analyzers/codecop-aa0139-donotassigntostringwithsmallercapacity), the message would be "Do not assign a text to a target with smaller size.".

**Comment by mazhelez** [comment]
Great suggestions but at this time we do not plan on working on improving code actions related to such issues.
